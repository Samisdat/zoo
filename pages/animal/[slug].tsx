    import React from 'react';
import { useRouter } from 'next/router'
import {getAnimals, getAnimal, Animal} from "pages/api/animals";
import renderToString from 'next-mdx-remote/render-to-string'
import hydrate from 'next-mdx-remote/hydrate'
    import Typography from "@material-ui/core/Typography";
    import CardContent from "@material-ui/core/CardContent";

export default function Tiere(props) {

    const router = useRouter()
    const { slug } = router.query

    const animal: Animal = props.animal as Animal;

    const image = props.animal.images[0];

    return (
        <React.Fragment>
            <Typography component="h1">
                {animal.title}
            </Typography>

            <img
                src={image}
                style={{
                    width: 300
                }}
            />

            <div>
                <a href={animal.wikipediaLink}>
                    {animal.wikipediaLink}
                </a>
            </div>
            <div>
                <a href={animal.iucnLink}>
                    {animal.iucnLink}
                </a>
            </div>
        </React.Fragment>
    );
}

export async function getStaticProps(context) {

    const animal = await getAnimal(context.params.slug);

    return {
        props: {
            animal:animal,
        }
    }
}

export async function getStaticPaths() {

    const animals = await getAnimals();

    const animalPaths = animals.map((animal:Animal)=>{
        return {
            params:{
                slug: animal.slug
            }
        }
    });

    console.log(animals)

    return {

        paths: animalPaths,

        fallback: false,
    }
}