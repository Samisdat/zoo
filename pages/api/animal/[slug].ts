import { NextApiRequest, NextApiResponse } from 'next'
import fs from "fs";
import path from "path";

export type Taxonomy = {
    class: string,
    order: string,
    family: string,
    species: string,
};

export type Animal = {
    slug: string,
    name: string,
    images: string[]
    taxonomy: Animal,

}

const dataDir = path.resolve(process.env.PWD, 'data/species');

export const getAnimal = async (slug:string):Promise<Animal> => {

    const filePath = path.resolve(dataDir, slug + '.json');

    const fileContent = await fs.readFileSync(filePath, {encoding:'utf8'});
    const fileJson = JSON.parse(fileContent);

    // @TODO validate

    return fileJson;
}


export const getAnimals = async ():Promise<Animal[]> => {

    const animals:Animal[] = [];

    let slugs = await fs.readdirSync(dataDir).map((file) => {
        return file.replace('.json', '');
    });

    for(const slug of slugs){

        const animal = await getAnimal(slug);

        animals.push(animal);
    }

    return animals;
}

export default async (req: NextApiRequest, res: NextApiResponse<Animal[]>) => {

    const animals = await getAnimals();

    res.status(200).json(animals);

}