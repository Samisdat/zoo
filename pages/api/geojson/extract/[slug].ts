import {NextApiRequest, NextApiResponse} from "next";
import fs from "fs";
import path from "path";
import {xmlTemplate} from "../../data/xml-template";

import {getSlug} from "../../../../helper/getSlug";

const { geoFromSVGXML } = require('svg2geojson');

export const getRectIds = (svg:string):string[] => {

    const pathIds:string[] = [];

    let pathRegEx = /<rect id="(.*?)"(?: serif:id="(.*?)")*/gm;

    let index = 0;

    let matches;

    while ((matches = pathRegEx.exec(svg)) !== null) {

        if (matches.index === pathRegEx.lastIndex) {
            pathRegEx.lastIndex++;
        }

        matches.forEach((match, groupIndex) => {
            if(1 === groupIndex){
                pathIds[index] = match;
                console.log(`Found match, group ${groupIndex}: ${match}`);
            }
            if(2 === groupIndex && undefined !== match){
                pathIds[index] = match;
                console.log(`Found match, group ${groupIndex}: ${match}`);
            }
        });

        index += 1;

    }

    return pathIds;

};

export default async (req: NextApiRequest, res: NextApiResponse<any>) => {

    const {
        query: {slug},
    } = req;



    const dataDir = path.resolve(process.env.PWD, 'pages/api/data');
    const combinedSvgPath = path.resolve(dataDir, 'combined.svg');
    const combinedSvg = fs.readFileSync(combinedSvgPath, {encoding:'utf8'});

    const groupRegEx = new RegExp(`<g id="${slug}"(.*?)>(.*?)<\/g>`, 'm')

    // @TODO debug
    // if slug !==
    if('zoomboxes' !== slug){
        res.status(400).json({
            error: `For now only slug zoomboxes is supported`
        });
    }


    // look into combined.svg
    // if no group with id=slug -> error
    if(false === groupRegEx.test(combinedSvg)){
        res.status(400).json({
            error: `No group id="${slug} could be found in ${combinedSvgPath}"`
        });
    }

    // create directory if not exits
    const dirForRequestSlug = path.resolve(dataDir, (slug as string));

    if (false === fs.existsSync(dirForRequestSlug)){
        fs.mkdirSync(dirForRequestSlug);
    }

    // create new svg only with group with id=slug
    const find =  groupRegEx.exec(combinedSvg);

    // add metainfo
    const dataSvg = xmlTemplate(find[2]);

    // write svg

    fs.writeFileSync(
        path.resolve(dirForRequestSlug, 'data.svg'),
        dataSvg,
        {encoding: 'utf8'}
    );

    const pathIds = getRectIds(dataSvg);

    geoFromSVGXML( dataSvg, (geoJson:any) => {


        for(let i = 0, x = geoJson.features.length; i < x; i += 1){

            const name =  pathIds[i];

            const slug = getSlug(name);

            geoJson.features[i].properties = {
                name,
                slug
            };

            console.log(geoJson.features[i].properties);

        }

        fs.writeFileSync(
            path.resolve(dirForRequestSlug, 'geo.json'),
            JSON.stringify(geoJson, null, 4),
            {encoding: 'utf8'}
        );

        res.status(200).json(geoJson);


    });


}