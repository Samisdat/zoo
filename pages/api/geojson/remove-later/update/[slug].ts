import { NextApiRequest, NextApiResponse } from 'next'

import path from 'path';
import fs from 'fs';

import urlSlug from 'url-slug'

const { geoFromSVGFile, geoFromSVGXML } = require('svg2geojson');

const metaInfo = '<MetaInfo xmlns="http://www.prognoz.ru"><Geo><GeoItem X="0" Y="0" Latitude="51.24177020918754" Longitude="7.105611562728882"/><GeoItem X="2550" Y="1994" Latitude="51.236776961813064" Longitude="7.115809321403503"/></Geo></MetaInfo>';

export default async (req: NextApiRequest, res: NextApiResponse<any[]>) => {

    const {
        query: { slug },
    } = req

    console.log(slug)

    const dataDir = path.resolve(process.env.PWD + '/pages/api/data');

    const json = JSON.parse(fs.readFileSync(dataDir + '/' + slug + '/data.json', {encoding: 'utf8'}));
    json.slug = slug;


    let svg = fs.readFileSync(dataDir + '/' + slug + '/data.svg', {encoding: 'utf8'});

    const regExMetaInfo = /<MetaInfo/;

    if(false === regExMetaInfo.test(svg)){

        const regEx = /<svg(.*?)>/;

        const updatedSvg = svg.replace(regEx, "<svg$1>" + metaInfo);

        fs.writeFileSync(dataDir + '/' + slug + '/data.svg', updatedSvg, {encoding: 'utf8'});

        svg = updatedSvg;

    }
    else {
        console.log('already contains MetaInfo');
    }

    geoFromSVGXML( svg, geoJson => {

        geoJson.features[0].properties = json;

        console.log(geoJson.features[0].properties, geoJson.features[0].type);

        fs.writeFileSync(dataDir + '/' + slug + '/geo.json', JSON.stringify(geoJson, null, 4), {encoding: 'utf8'});

    });

    fs.writeFileSync(dataDir + '/' + slug + '/data.json', JSON.stringify(json, null, 4), {encoding: 'utf8'});

    res.status(200).json(json);

}