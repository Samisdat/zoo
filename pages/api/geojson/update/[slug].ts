import { NextApiRequest, NextApiResponse } from 'next'

import path from 'path';
import fs from 'fs';

const { geoFromSVGXML } = require('svg2geojson');

export const addMetaInfo = (svg:string):string => {

    const regExMetaInfo = /<MetaInfo/m;

    if(true === regExMetaInfo.test(svg)){
        return svg;
    }

    const metaInfo = '<MetaInfo xmlns="http://www.prognoz.ru"><Geo><GeoItem X="0" Y="0" Latitude="51.24177020918754" Longitude="7.105611562728882"/><GeoItem X="2550" Y="1994" Latitude="51.236776961813064" Longitude="7.115809321403503"/></Geo></MetaInfo>';

    const regEx = /<svg(.*?)>/;

    const updatedSvg = svg.replace(regEx, "<svg$1>" + metaInfo);

    return updatedSvg;

}

export default async (req: NextApiRequest, res: NextApiResponse<any>) => {

    const {
        query: { slug },
    } = req;

    if(undefined === slug){
        res.status(400).json({msg:'slug not defined'});
        return;
    }
    const dataDir = path.resolve(process.env.PWD + '/pages/api/data');

    const json = JSON.parse(fs.readFileSync(dataDir + '/' + slug + '/data.json', {encoding: 'utf8'}));
    json.slug = slug;

    let svg = fs.readFileSync(dataDir + '/' + slug + '/data.svg', {encoding: 'utf8'});

    svg = addMetaInfo(svg);

    fs.writeFileSync(dataDir + '/' + slug + '/data.svg', svg, {encoding: 'utf8'});

    geoFromSVGXML( svg, (geoJson:any) => {

        geoJson.features[0].properties = json;

        console.log(geoJson.features[0].properties, geoJson.features[0].type);

        fs.writeFileSync(dataDir + '/' + slug + '/geo.json', JSON.stringify(geoJson, null, 4), {encoding: 'utf8'});

    });

    fs.writeFileSync(dataDir + '/' + slug + '/data.json', JSON.stringify(json, null, 4), {encoding: 'utf8'});

    res.status(200).json(json);

}