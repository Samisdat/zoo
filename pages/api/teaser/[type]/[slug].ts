import { NextApiRequest, NextApiResponse } from 'next'
import {getSlug} from "helper/getSlug";
import fs from "fs";
import path from "path";
import {getAnimals, getAnimal} from "../../animals";
import {getFacilities} from "../../facilities";
const frontmatter = require('@github-docs/frontmatter')

export interface TeaserInterface {
    href: string;
    title: string;
    image: string;
}

const dataDir = path.resolve(process.env.PWD + '/data/markdown');

export default async (req: NextApiRequest, res: NextApiResponse<TeaserInterface[]>) => {

    let {
        query: {type,slug},
    } = req;

    // @TODO what about type or slug is string[]

    type = type as string;
    slug = slug as string;

    const teasers :TeaserInterface[] = [];

    console.log(type, slug)

    if('animal' === type){

        const animals = await getAnimals();

        console.log(animals);

        const animal = animals.find((animal)=>{
            if(slug === animal.facility){
                return true;
            }

            return false;
        });

        const href = `/tiere/${animal.slug}`;

        let image = undefined;

        if(undefined !== animal.images && animal.images.length > 0){
            image = animal.images[0];
        }

        const teaser:TeaserInterface = {
            image: image,
            title: animal.title,
            href: href,
        };

        teasers.push(teaser);

    }
    else if('facility' === type){

        const facilities = await getFacilities();


        const facility = facilities.find((facility)=>{
            if(slug === facility.slug){
                return true;
            }

            return false;
        });

        if('food' === facility.type){

            const href = `/anlagen/${facility.slug}`;

            const teaser:TeaserInterface = {
                image: facility.images[0],
                title: facility.title,
                href: href,
            };

            teasers.push(teaser);

        }
        else if(
            'single-enclosure' === facility.type ||
            'shared-enclosure' === facility.type
        ){

        for(let i = 0, x = facility.animals.length; i < x; i += 1){

            const animalSlug = facility.animals[i];

            const animal = await getAnimal(animalSlug);

            const href = `/tiere/${animal.slug}`;

            let image = undefined;

            if(undefined !== animal.images && animal.images.length > 0){
                image = animal.images[0];
            }

            const teaser:TeaserInterface = {
                image: image,
                title: animal.title,
                href: href,
            };

            teasers.push(teaser);

        }
        }

    }

    res.status(200).json(teasers);

}