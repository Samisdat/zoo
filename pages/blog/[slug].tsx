import React from 'react';
import {Feature} from "geojson";
import {getFullGeoJson} from "../api/geojson/list";
import path from "path";
import fs from "fs";
import Moment from "react-moment";
const frontmatter = require('@github-docs/frontmatter')

const ReactMarkdown = require('react-markdown')
const gfm = require('remark-gfm')

export default function Gehege(props) {

    return (
        <React.Fragment>
            <h1>{props.title}</h1>
            <h2><Moment format="DD.MM.YYYY" date={props.date} /></h2>
            <ReactMarkdown plugins={[gfm]} children={props.content} />
        </React.Fragment>
    );
}

export async function getStaticProps(context) {

    const slugParam = context.params.slug

    const dataDir = path.resolve(process.env.PWD, 'data/markdown/news');

    const newsFilePath = path.resolve(dataDir, slugParam + '.md');

    const newsFileContent = fs.readFileSync(newsFilePath, {encoding:'utf8'});

    const { data, content, errors } = frontmatter(newsFileContent);

    const { title, slug, date, animal, enclosure } = data;



    return {
        props: {
            title,
            slug,
            date: date.toISOString(),
            animal,
            enclosure,
            content
        }
    }
}


export async function getStaticPaths() {


    const dataDir = path.resolve(process.env.PWD, 'data/markdown/news');

    const newsPostFiles = fs.readdirSync(dataDir);

    let newsSlugs = [];

    for(const newsPostFile of newsPostFiles) {

        if ('.DS_Store' === newsPostFile) {
            continue;
        }

        const newsFilePath = path.resolve(dataDir, newsPostFile);

        if (false === fs.existsSync(newsFilePath)) {
            continue;
        }

        const newsFileContent = fs.readFileSync(newsFilePath, {encoding: 'utf8'});

        const {data} = frontmatter(newsFileContent);
        const {slug} = data;

        newsSlugs.push({
            params:{
                slug: slug + ''
            }
        });

    }

    console.log(newsSlugs)

    return {

        paths: newsSlugs,

        fallback: false,
    }
}