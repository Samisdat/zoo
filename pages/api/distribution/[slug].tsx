import { NextApiRequest, NextApiResponse } from 'next'
import {getDistribution} from "../../tiere/[slug]";

export default async (req: NextApiRequest, res: NextApiResponse<any>) => {

    let {
        query: {slug},
    } = req;

    // @TODO what about type or slug is string[]

    slug = slug as string;

    const distribution = await getDistribution(slug)

    res.status(200).json(distribution);

}