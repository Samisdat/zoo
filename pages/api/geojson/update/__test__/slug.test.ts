import {addMetaInfo} from "../[slug]";

import { apiResolver } from 'next/dist/next-server/server/api-utils'
import handler from "../[slug]";

import { createMocks } from 'node-mocks-http';
import handleUpdate from '../[slug]';

describe('geojson/update endpoint', () => {

    it('returns status 200 and an object for valid slug', async () => {
        const { req, res } = createMocks({
            method: 'GET',
            query: {
                slug: 'aussengrenze',
            },
        });

        await handleUpdate(req, res);

        expect(res._getStatusCode()).toBe(200);

        expect(JSON.parse(res._getData())).toEqual(
            expect.objectContaining(
                {"fill": "#B6E2B6", "name": "AuÃŸengrenze", "slug": "aussengrenze", "zIndex": 1}
            ),
        );
    });
});

describe('update/create geojson ', () => {

    it('adds meta info to svg if not yet present - multiline', ()=>{

        const svgWithoutMetaInfo = `<svg width="100%" height="100%" viewBox="0 0 2550 1994" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" xmlns:serif="http://www.serif.com/" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;">
            <g id="Zoo"></g>
        </svg>`;

        expect(svgWithoutMetaInfo).not.toEqual(expect.stringContaining('MetaInfo'));

        const svg = addMetaInfo(svgWithoutMetaInfo);

        expect(svg).toEqual(expect.stringContaining('MetaInfo'));
        expect(svg).toMatchSnapshot();

    });

    it('adds meta info to svg if not yet present - singleline', ()=>{

        const svgWithoutMetaInfo = `<svg width="100%" height="100%" viewBox="0 0 2550 1994" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" xmlns:serif="http://www.serif.com/" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="Zoo"></g></svg>`;

        expect(svgWithoutMetaInfo).not.toEqual(expect.stringContaining('MetaInfo'));

        const svg = addMetaInfo(svgWithoutMetaInfo);

        expect(svg).toEqual(expect.stringContaining('MetaInfo'));
        expect(svg).toMatchSnapshot();

    });

    it('leave svg untouched if meta info is already present - multiline', ()=>{

        const svgWithoutMetaInfo = `<svg width="100%" height="100%" viewBox="0 0 2550 1994" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" xmlns:serif="http://www.serif.com/" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;">
            <MetaInfo xmlns="http://www.prognoz.ru">
                <Geo>
                    <GeoItem X="0" Y="0" Latitude="51.24177020918754" Longitude="7.105611562728882"/>
                    <GeoItem X="2550" Y="1994" Latitude="51.236776961813064" Longitude="7.115809321403503"/>
                </Geo>
            </MetaInfo>
            <g id="Zoo"></g>
        </svg>`;

        expect(svgWithoutMetaInfo).toEqual(expect.stringContaining('MetaInfo'));

        const svg = addMetaInfo(svgWithoutMetaInfo);

        expect(svg).toEqual(expect.stringContaining('MetaInfo'));
        expect(svg).toMatchSnapshot();


    });

    it('leave svg untouched if meta info is already present - singleline', ()=>{

        const svgWithoutMetaInfo = `<svg width="100%" height="100%" viewBox="0 0 2550 1994" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" xmlns:serif="http://www.serif.com/" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><MetaInfo xmlns="http://www.prognoz.ru"><Geo><GeoItem X="0" Y="0" Latitude="51.24177020918754" Longitude="7.105611562728882"/><GeoItem X="2550" Y="1994" Latitude="51.236776961813064" Longitude="7.115809321403503"/></Geo></MetaInfo><g id="Zoo"></g></svg>`;

        expect(svgWithoutMetaInfo).toEqual(expect.stringContaining('MetaInfo'));

        const svg = addMetaInfo(svgWithoutMetaInfo);

        expect(svg).toEqual(expect.stringContaining('MetaInfo'));
        expect(svg).toMatchSnapshot();

    });

});



