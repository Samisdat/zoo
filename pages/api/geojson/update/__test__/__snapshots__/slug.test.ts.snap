// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update/create geojson  adds meta info to svg if not yet present - multiline 1`] = `
"<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\" xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;\\"><MetaInfo xmlns=\\"http://www.prognoz.ru\\"><Geo><GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/><GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/></Geo></MetaInfo>
            <g id=\\"Zoo\\"></g>
        </svg>"
`;

exports[`update/create geojson  adds meta info to svg if not yet present - singleline 1`] = `"<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\" xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;\\"><MetaInfo xmlns=\\"http://www.prognoz.ru\\"><Geo><GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/><GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/></Geo></MetaInfo><g id=\\"Zoo\\"></g></svg>"`;

exports[`update/create geojson  leave svg untouched if meta info is already present - multiline 1`] = `
"<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\" xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;\\">
            <MetaInfo xmlns=\\"http://www.prognoz.ru\\">
                <Geo>
                    <GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/>
                    <GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/>
                </Geo>
            </MetaInfo>
            <g id=\\"Zoo\\"></g>
        </svg>"
`;

exports[`update/create geojson  leave svg untouched if meta info is already present - singleline 1`] = `"<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\" xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;\\"><MetaInfo xmlns=\\"http://www.prognoz.ru\\"><Geo><GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/><GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/></Geo></MetaInfo><g id=\\"Zoo\\"></g></svg>"`;
